src/common/types/uuid.cpp

```diff
@@ -0,0 +1,89 @@
+#include "common/types/uuid.h"
+
+namespace kuzu {
+namespace common {
+
+bool UUID::fromString(std::string str, int128_t& result) {
+    auto hex2Char = [](char ch) -> unsigned char {
+        if (ch >= '0' && ch <= '9') {
+            return ch - '0';
+        }
+        if (ch >= 'a' && ch <= 'f') {
+            return 10 + ch - 'a';
+        }
+        if (ch >= 'A' && ch <= 'F') {
+            return 10 + ch - 'A';
+        }
+        return 0;
+    };
+    auto isHex = [](char ch) -> bool {
+        return (ch >= '0' && ch <= '9') || (ch >= 'a' && ch <= 'f') || (ch >= 'A' && ch <= 'F');
+    };
+
+    if (str.empty()) {
+        return false;
+    }
+    int hasBraces = 0;
+    if (str.front() == '{') {
+        hasBraces = 1;
+    }
+    if (hasBraces && str.back() != '}') {
+        return false;
+    }
+
+    result.low = 0;
+    result.high = 0;
+    size_t count = 0;
```

Comment:
We never use size_t in the codebase since it is platform dependent, use uint32/uint64_t instead.